### PR TITLE
Add ACAS support

### PR DESCRIPTION
**Pull request by tuftedocelot**
Definitely not ready yet. I need to find a source of actual ACAS messages versus reading straight from the specs (not pretty)

### CODE REVIEW

**Comment by junzis** [comment]
I updated the structure and functions. 

`acas.py` is meant for DF=0/16. So I moved some functions to BDS61 `bds61_st2.py`.

I also consolidated some of the functions, and still need to properly test the code before merging. But it looks quite good already.

**Comment by tuftedocelot** [comment]
:+1: 

Thanks for the ARA material. I couldn't decide how to return/present that info back.

When it comes to tests, are there any good ADS-B/Mode S message repositories already out there? I checked opensky's Java library and there aren't even any tests in there to borrow valid messages from. The only other thing I could think of was to log any/all messages that I can receive and go spelunking for the relevant message types

**Comment by junzis** [comment]
> When it comes to tests, are there any good ADS-B/Mode S message repositories already out there? I checked opensky's Java library and there aren't even any tests in there to borrow valid messages from. The only other thing I could think of was to log any/all messages that I can receive and go spelunking for the relevant message types

@xoolive, do you have any hand-picked BDS61 messages from your recent paper for a proper unit test?

**Comment by Flyer350** [comment]
I don't understand the link between ACAS messages and BDS61? (extended squitter emergency/priority status)

**Comment by junzis** [comment]
> I don't understand the link between ACAS messages and BDS61? (extended squitter emergency/priority status)

It's been a while. I think it was about the coordinated reply in DF=16, which has (almost) the same structure as BDS 6,1 (subtype 2)

**Comment by junzis** [comment]
The MV field in DF=16 can be any GICB data. This makes the decoding very complicated. 

It is similar to what we did with BDS inferring, except that now we are probably getting extended squitter data, and this is very hard to inference without knowing the data selector code.

File: tests/test_acas.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "8526D57E4D963C92CDEE1B6C7C49",
     "802A613C93F65A7FF803A51B5ADB",
     "85B6C54279B67BE2A0001998FFDA",
-    "851944F15881648338D1AF4B7A27",
+    "851944F1 5881648338D1AF 4B7A27",
```
Comment (xoolive) [inline tests/test_acas.py:28]: Why spaces??

**Comment by tedder** [comment]
I can run it against DF16 messages, though they all return False from isACAS. I note it says it's for `3,1`, but then the string appears to be for `3,0`:
```
    :return: if VDS is 3,1
...
    vds = mv[0:8]
    if vds == "00110000":
```

Looks like the messages I'm seeing with DF16 are UDS 5,8 and 5,9. Not sure if it'd be helpful to paste them in here, but I thought I'd at least indicate I'd tested isACAS.

**Comment by junzis** [comment]
I am going to pull this into a separate branch and then work from there.

**Pull request closed without merging**